### ISSUE TITLE
[NvFuser] The JIT model with `mean` will crash when computing the gradients on nvfuser

### ISSUE THREAD

**Issue by payphon** [pytorch/pytorch#86251]
### 🐛 Describe the bug

The JIT model with `mean` will crash when computing the gradients on cuda.
It seems that it frees an invalid pointer

```py
import torch

class M(torch.nn.Module):
    def __init__(self):
        super().__init__()
        self.dim = -1
        self.keepdim = False

    def forward(self, input):
        dim = self.dim
        keepdim = self.keepdim
        input = torch.sub(input, torch.tensor(9, dtype=torch.float32, device='cuda'))
        fn_res = torch.mean(input, dim, keepdim=keepdim, )
        fn_res = torch.sub(fn_res, torch.tensor(-3, dtype=torch.float32, device='cuda'))
        return fn_res

fn = M().to('cuda')
torch.random.manual_seed(56288)
inp = torch.empty([1, 64], dtype=torch.float32, memory_format=torch.contiguous_format)
inp.uniform_(-128, 63)

def clone(tensor):
    return tensor.clone().to('cuda').requires_grad_()

jit_fn = torch.jit.trace(fn, clone(inp))

from torch.autograd.functional import jacobian
jit_fn(clone(inp))

jacobian(jit_fn, (clone(inp), ))

jacobian(jit_fn, (clone(inp), ), vectorize=True, strategy='forward-mode')
```

```
free(): invalid pointer
[1]    3996840 IOT instruction (core dumped)  python mean.py
```

### Versions

```
PyTorch version: 1.13.0.dev20221001
Is debug build: False
CUDA used to build PyTorch: 11.7
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.1 LTS (x86_64)
GCC version: (Ubuntu 11.2.0-19ubuntu1) 11.2.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.35

Python version: 3.9.13 (main, Aug 25 2022, 23:26:10)  [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-5.19.5-051905-generic-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: 11.7.99
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 3060
Nvidia driver version: 515.65.01
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.23.3
[pip3] torch==1.13.0.dev20221001
[pip3] torchaudio==0.13.0.dev20221001
[pip3] torchvision==0.14.0.dev20221001
[conda] blas                      1.0                         mkl  
[conda] mkl                       2021.4.0           h06a4308_640  
[conda] mkl-service               2.4.0            py39h7f8727e_0  
[conda] mkl_fft                   1.3.1            py39hd3c417c_0  
[conda] mkl_random                1.2.2            py39h51133e4_0  
[conda] numpy                     1.23.3                   pypi_0    pypi
[conda] numpy-base                1.23.1           py39ha15fc14_0  
[conda] pytorch                   1.13.0.dev20221001 py3.9_cuda11.7_cudnn8.5.0_0    pytorch-nightly
[conda] pytorch-cuda              11.7                 h67b0de4_0    pytorch-nightly
[conda] pytorch-mutex             1.0                        cuda    pytorch-nightly
[conda] torch                     1.13.0.dev20220923+cu116          pypi_0    pypi
[conda] torchaudio                0.13.0.dev20220923+cu116          pypi_0    pypi
[conda] torchvision               0.14.0.dev20220923+cu116          pypi_0    pypi
```

**Comment by payphon** [comment]
It seems that it only appears with NvFuser since when I use ` PYTORCH_JIT_USE_NNC_NOT_NVFUSER=1`, it will not crash

**Comment by payphon** [comment]
Hi @kevinstephano, could you please take a look at this issue since I think it is crash problem related to NvFuser

**Comment by kevinstephano** [comment]
I am not sure if nvFuser is responsible or if this is a general TorchScript Executor issue as the stacktrace results do not point directly at nvFuser and nvFuser does not allocate memory.

```
Thread 1 "python" received signal SIGABRT, Aborted.
0x00007ffff7c9000b in raise () from /usr/lib/x86_64-linux-gnu/libc.so.6
(gdb) where
#0  0x00007ffff7c9000b in raise () from /usr/lib/x86_64-linux-gnu/libc.so.6
#1  0x00007ffff7c6f859 in abort () from /usr/lib/x86_64-linux-gnu/libc.so.6
#2  0x00007ffff7cda26e in ?? () from /usr/lib/x86_64-linux-gnu/libc.so.6
#3  0x00007ffff7ce22fc in ?? () from /usr/lib/x86_64-linux-gnu/libc.so.6
#4  0x00007ffff7ce3b2c in ?? () from /usr/lib/x86_64-linux-gnu/libc.so.6
#5  0x00007fff9f71c29a in c10::detail::ListImpl::~ListImpl() () from /workspace/pytorch/torch/lib/libtorch_cpu.so
#6  0x00007fffa2e5d887 in std::_Sp_counted_ptr_inplace<torch::jit::(anonymous namespace)::DifferentiableGraphBackward, std::allocator<torch::jit::(anonymous namespace)::DifferentiableGraphBackward>, (__gnu_cxx::_Lock_policy)2>::_M_dispose() ()
   from /workspace/pytorch/torch/lib/libtorch_cpu.so
#7  0x00007fffa1e80ca6 in torch::autograd::Node::~Node() () from /workspace/pytorch/torch/lib/libtorch_cpu.so
#8  0x00007fffa2a729f8 in torch::autograd::AutogradMeta::~AutogradMeta() () from /workspace/pytorch/torch/lib/libtorch_cpu.so
#9  0x00007fffa2ac4196 in torch::autograd::DifferentiableViewMeta::~DifferentiableViewMeta() () from /workspace/pytorch/torch/lib/libtorch_cpu.so
#10 0x00007fff7aa90921 in c10::TensorImpl::~TensorImpl() () from /workspace/pytorch/torch/lib/libc10.so
#11 0x00007fff7aa90a8d in c10::TensorImpl::~TensorImpl() () from /workspace/pytorch/torch/lib/libc10.so
#12 0x00007fffa9778d88 in THPVariable_clear(THPVariable*) () from /workspace/pytorch/torch/lib/libtorch_python.so
#13 0x00007fffa97790d5 in THPVariable_subclass_dealloc(_object*) () from /workspace/pytorch/torch/lib/libtorch_python.so
#14 0x0000555555663660 in _Py_Dealloc (op=<optimized out>) at /opt/conda/conda-bld/python-split_1648465063888/work/Objects/object.c:2215
```

**Comment by payphon** [comment]
Thanks for your help and reply!

Maybe we need a TorchScript developer to debug it :)
